src/command_modules/azure-cli-cdn/azure/cli/command_modules/cdn/custom.py

```diff
@@ -0,0 +1,129 @@
+        origin_host_header=origin_host_header,
+        origin_path=origin_path,
+        content_types_to_compress=content_types_to_compress,
+        is_compression_enabled=is_compression_enabled,
+        is_http_allowed=is_http_allowed,
+        is_https_allowed=is_https_allowed,
+        query_string_caching_behavior=query_string_caching_behavior,
+        tags=tags
+    )
+    _update_mapper(instance, params, [
+        'origin_host_header',
+        'origin_path',
+        'content_types_to_compress',
+        'is_compression_enabled',
+        'is_http_allowed',
+        'is_https_allowed',
+        'query_string_caching_behavior',
+        'tags'
+    ])
+    return params
+update_endpoint.__doc__ = EndpointUpdateParameters.__doc__
+
+
+def create_endpoint(client, resource_group_name, profile_name, name, origins, location=None,
+                    origin_host_header=None, origin_path=None, content_types_to_compress=None,
+                    is_compression_enabled=None, is_http_allowed=None, is_https_allowed=None,
+                    query_string_caching_behavior=QueryStringCachingBehavior.ignore_query_string.
+                    value, tags=None):
+    is_compression_enabled = False if is_compression_enabled is None else is_compression_enabled
+    is_http_allowed = True if is_http_allowed is None else is_http_allowed
+    is_https_allowed = True if is_https_allowed is None else is_https_allowed
+    endpoint = Endpoint(location,
+                        origins,
+                        origin_host_header=origin_host_header,
+                        origin_path=origin_path,
+                        content_types_to_compress=content_types_to_compress,
+                        is_compression_enabled=is_compression_enabled,
+                        is_http_allowed=is_http_allowed,
+                        is_https_allowed=is_https_allowed,
+                        query_string_caching_behavior=query_string_caching_behavior,
+                        tags=tags)
+    return client.endpoints.create(resource_group_name, profile_name, name, endpoint)
+create_endpoint.__doc__ = Endpoint.__doc__
+
+
+def create_custom_domain(client, resource_group_name, profile_name, endpoint_name, name, hostname,
+                         location=None, tags=None):
+    """
+    Creates a new custom domain within an endpoint.
+    :param client: CdnManagementClint
+    :param resource_group_name: Name of resource group.
+    :param profile_name: Name of the CDN profile which is unique within the resource group.
+    :param endpoint_name: Name of the endpoint under the profile which is unique globally.
+    :param name: Name of the custom domain within an endpoint.
+    :param hostname: The host name of the custom domain. Must be a domain name.
+    :param location: Location of the profile. Defaults to the location of the resource group.
+    :param tags: Resource manager tags
+    :return: The newly created CDN profile
+    :rtype: Profile
```

Comment:
Please move this into help.py.  Docstrings on custom commands is the only discouraged way of registering parameter help.  The reason is you get more benefit from having it in help.py in exchange for the same downside you buy into by *not* having it in params.py.